### PR TITLE
fix: upgrade all packages to latest to fix vulnerabilities

### PR DESCRIPTION
**Pull request by saikumarrs**
## PR Description

I've upgraded all the package dependencies to the latest versions to fix vulnerabilities.
Additionally, all the dependencies in the sample apps are also updated to latest.

- Fixed issues in GitHub workflows where the dependencies were not correctly established.
- Increased the timeout b/w tests in the sanity test suite to accommodate testing on mobile devices.

Synk CLI report after the changes:
```
snyk test --all-projects                    
✗ 9/26 potential projects failed to get dependencies.
/Volumes/Workspace/Repositories/rudder-sdk-js/packages/analytics-js/project.json:
  No valid Dotnet target framework found
/Volumes/Workspace/Repositories/rudder-sdk-js/packages/analytics-js-common/project.json:
  No valid Dotnet target framework found
/Volumes/Workspace/Repositories/rudder-sdk-js/packages/analytics-js-cookies/project.json:
  No valid Dotnet target framework found
/Volumes/Workspace/Repositories/rudder-sdk-js/packages/analytics-js-integrations/project.json:
  No valid Dotnet target framework found
/Volumes/Workspace/Repositories/rudder-sdk-js/packages/analytics-js-plugins/project.json:
  No valid Dotnet target framework found
/Volumes/Workspace/Repositories/rudder-sdk-js/packages/analytics-js-service-worker/project.json:
  No valid Dotnet target framework found
/Volumes/Workspace/Repositories/rudder-sdk-js/packages/analytics-v1.1/project.json:
  No valid Dotnet target framework found
/Volumes/Workspace/Repositories/rudder-sdk-js/packages/loading-scripts/project.json:
  No valid Dotnet target framework found
/Volumes/Workspace/Repositories/rudder-sdk-js/packages/sanity-suite/project.json:
  No valid Dotnet target framework found

Testing /Volumes/Workspace/Repositories/rudder-sdk-js...

Organization:      rudder-qa
Package manager:   npm
Target file:       package.json
Project name:      @rudderstack/analytics-js-monorepo
Open source:       no
Project path:      /Volumes/Workspace/Repositories/rudder-sdk-js
Licenses:          enabled

✔ Tested 89 dependencies for known issues, no vulnerable paths found.

Next steps:
- Run `snyk monitor` to be notified about new related vulnerabilities.
- Run `snyk test` as part of your CI/test.

-------------------------------------------------------

Testing /Volumes/Workspace/Repositories/rudder-sdk-js...

Organization:      rudder-qa
Package manager:   npm
Target file:       packages/analytics-js/package.json
Project name:      @rudderstack/analytics-js
Open source:       no
Project path:      /Volumes/Workspace/Repositories/rudder-sdk-js
Licenses:          enabled

✔ Tested 23 dependencies for known issues, no vulnerable paths found.

Next steps:
- Run `snyk monitor` to be notified about new related vulnerabilities.
- Run `snyk test` as part of your CI/test.

-------------------------------------------------------

Testing /Volumes/Workspace/Repositories/rudder-sdk-js...

Organization:      rudder-qa
Package manager:   npm
Target file:       packages/analytics-js-common/package.json
Project name:      @rudderstack/analytics-js-common
Open source:       no
Project path:      /Volumes/Workspace/Repositories/rudder-sdk-js
Licenses:          enabled

✔ Tested 17 dependencies for known issues, no vulnerable paths found.

Next steps:
- Run `snyk monitor` to be notified about new related vulnerabilities.
- Run `snyk test` as part of your CI/test.

-------------------------------------------------------

Testing /Volumes/Workspace/Repositories/rudder-sdk-js...

Organization:      rudder-qa
Package manager:   npm
Target file:       packages/analytics-js-cookies/package.json
Project name:      @rudderstack/analytics-js-cookies
Open source:       no
Project path:      /Volumes/Workspace/Repositories/rudder-sdk-js
Licenses:          enabled

✔ Tested 18 dependencies for known issues, no vulnerable paths found.

Next steps:
- Run `snyk monitor` to be notified about new related vulnerabilities.
- Run `snyk test` as part of your CI/test.

-------------------------------------------------------

Testing /Volumes/Workspace/Repositories/rudder-sdk-js...

Organization:      rudder-qa
Package manager:   npm
Target file:       packages/analytics-js-integrations/package.json
Project name:      @rudderstack/analytics-js-integrations
Open source:       no
Project path:      /Volumes/Workspace/Repositories/rudder-sdk-js
Licenses:          enabled

✔ Tested 32 dependencies for known issues, no vulnerable paths found.

Next steps:
- Run `snyk monitor` to be notified about new related vulnerabilities.
- Run `snyk test` as part of your CI/test.

-------------------------------------------------------

Testing /Volumes/Workspace/Repositories/rudder-sdk-js...

Organization:      rudder-qa
Package manager:   npm
Target file:       packages/analytics-js-plugins/package.json
Project name:      @rudderstack/analytics-js-plugins
Open source:       no
Project path:      /Volumes/Workspace/Repositories/rudder-sdk-js
Licenses:          enabled

✔ Tested 21 dependencies for known issues, no vulnerable paths found.

Next steps:
- Run `snyk monitor` to be notified about new related vulnerabilities.
- Run `snyk test` as part of your CI/test.

-------------------------------------------------------

Testing /Volumes/Workspace/Repositories/rudder-sdk-js...

Organization:      rudder-qa
Package manager:   npm
Target file:       packages/analytics-js-service-worker/package.json
Project name:      @rudderstack/analytics-js-service-worker
Open source:       no
Project path:      /Volumes/Workspace/Repositories/rudder-sdk-js
Licenses:          enabled

✔ Tested 62 dependencies for known issues, no vulnerable paths found.

Next steps:
- Run `snyk monitor` to be notified about new related vulnerabilities.
- Run `snyk test` as part of your CI/test.

-------------------------------------------------------

Testing /Volumes/Workspace/Repositories/rudder-sdk-js...

Organization:      rudder-qa
Package manager:   npm
Target file:       packages/analytics-v1.1/package.json
Project name:      rudder-sdk-js
Open source:       no
Project path:      /Volumes/Workspace/Repositories/rudder-sdk-js
Licenses:          enabled

✔ Tested 18 dependencies for known issues, no vulnerable paths found.

Next steps:
- Run `snyk monitor` to be notified about new related vulnerabilities.
- Run `snyk test` as part of your CI/test.

-------------------------------------------------------

Testing /Volumes/Workspace/Repositories/rudder-sdk-js...

Organization:      rudder-qa
Package manager:   npm
Target file:       packages/loading-scripts/package.json
Project name:      @rudderstack/analytics-js-loading-scripts
Open source:       no
Project path:      /Volumes/Workspace/Repositories/rudder-sdk-js
Licenses:          enabled

✔ Tested 24 dependencies for known issues, no vulnerable paths found.

Next steps:
- Run `snyk monitor` to be notified about new related vulnerabilities.
- Run `snyk test` as part of your CI/test.

-------------------------------------------------------

Testing /Volumes/Workspace/Repositories/rudder-sdk-js...

Organization:      rudder-qa
Package manager:   npm
Target file:       packages/sanity-suite/package.json
Project name:      @rudderstack/analytics-js-sanity-suite
Open source:       no
Project path:      /Volumes/Workspace/Repositories/rudder-sdk-js
Licenses:          enabled

✔ Tested 27 dependencies for known issues, no vulnerable paths found.

Next steps:
- Run `snyk monitor` to be notified about new related vulnerabilities.
- Run `snyk test` as part of your CI/test.

-------------------------------------------------------

Testing /Volumes/Workspace/Repositories/rudder-sdk-js...

Organization:      rudder-qa
Package manager:   npm
Target file:       examples/angular/sample-app/package-lock.json
Project name:      sample-app
Open source:       no
Project path:      /Volumes/Workspace/Repositories/rudder-sdk-js
Licenses:          enabled

✔ Tested 12 dependencies for known issues, no vulnerable paths found.

Next steps:
- Run `snyk monitor` to be notified about new related vulnerabilities.
- Run `snyk test` as part of your CI/test.

-------------------------------------------------------

Testing /Volumes/Workspace/Repositories/rudder-sdk-js...

Tested 1311 dependencies for known issues, found 2 issues, 2 vulnerable paths.


Issues with no direct upgrade or patch:
  ✗ Missing Release of Resource after Effective Lifetime [Medium Severity][https://security.snyk.io/vuln/SNYK-JS-INFLIGHT-6095116] in inflight@1.0.6
    introduced by gatsby@5.13.7 > @graphql-codegen/typescript@2.8.8 > @graphql-codegen/visitor-plugin-common@2.13.8 > @graphql-tools/relay-operation-optimizer@6.5.18 > @ardatan/relay-compiler@12.0.0 > glob@7.2.3 > inflight@1.0.6
  No upgrade or patch available


License issues:

  ✗ MPL-2.0 license [Medium Severity][https://snyk.io/vuln/snyk:lic:npm:axe-core:MPL-2.0] in axe-core@4.10.0
    introduced by gatsby@5.13.7 > eslint-plugin-jsx-a11y@6.10.0 > axe-core@4.10.0



Organization:      rudder-qa
Package manager:   npm
Target file:       examples/gatsby/sample-gatsby-plugin-usage/package-lock.json
Project name:      sample-using-gatsby-plugin
Open source:       no
Project path:      /Volumes/Workspace/Repositories/rudder-sdk-js
Licenses:          enabled

-------------------------------------------------------

Testing /Volumes/Workspace/Repositories/rudder-sdk-js...

Tested 1295 dependencies for known issues, found 2 issues, 2 vulnerable paths.


Issues with no direct upgrade or patch:
  ✗ Missing Release of Resource after Effective Lifetime [Medium Severity][https://security.snyk.io/vuln/SNYK-JS-INFLIGHT-6095116] in inflight@1.0.6
    introduced by gatsby@5.13.7 > @graphql-codegen/typescript@2.8.8 > @graphql-codegen/visitor-plugin-common@2.13.8 > @graphql-tools/relay-operation-optimizer@6.5.18 > @ardatan/relay-compiler@12.0.0 > glob@7.2.3 > inflight@1.0.6
  No upgrade or patch available


License issues:

  ✗ MPL-2.0 license [Medium Severity][https://snyk.io/vuln/snyk:lic:npm:axe-core:MPL-2.0] in axe-core@4.10.0
    introduced by gatsby@5.13.7 > eslint-plugin-jsx-a11y@6.10.0 > axe-core@4.10.0



Organization:      rudder-qa
Package manager:   npm
Target file:       examples/gatsby/sample-gatsby-site/package-lock.json
Project name:      sample-gatsby-site
Open source:       no
Project path:      /Volumes/Workspace/Repositories/rudder-sdk-js
Licenses:          enabled

-------------------------------------------------------

Testing /Volumes/Workspace/Repositories/rudder-sdk-js...

Organization:      rudder-qa
Package manager:   npm
Target file:       examples/serverless/cloudflare-worker/package-lock.json
Project name:      example-cloudflare-worker
Open source:       no
Project path:      /Volumes/Workspace/Repositories/rudder-sdk-js
Licenses:          enabled

✔ Tested 8 dependencies for known issues, no vulnerable paths found.

Next steps:
- Run `snyk monitor` to be notified about new related vulnerabilities.
- Run `snyk test` as part of your CI/test.

-------------------------------------------------------

Testing /Volumes/Workspace/Repositories/rudder-sdk-js...

Organization:      rudder-qa
Package manager:   npm
Target file:       examples/serverless/vercel-edge/package-lock.json
Project name:      example-vercel-edge
Open source:       no
Project path:      /Volumes/Workspace/Repositories/rudder-sdk-js
Licenses:          enabled

✔ Tested 143 dependencies for known issues, no vulnerable paths found.

Next steps:
- Run `snyk monitor` to be notified about new related vulnerabilities.
- Run `snyk test` as part of your CI/test.

-------------------------------------------------------

Testing /Volumes/Workspace/Repositories/rudder-sdk-js...

Organization:      rudder-qa
Package manager:   composer
Target file:       examples/symfony/sample/composer.lock
Project name:      sample
Open source:       no
Project path:      /Volumes/Workspace/Repositories/rudder-sdk-js
Licenses:          enabled

✔ Tested 105 dependencies for known issues, no vulnerable paths found.

Next steps:
- Run `snyk monitor` to be notified about new related vulnerabilities.
- Run `snyk test` as part of your CI/test.

-------------------------------------------------------

Testing /Volumes/Workspace/Repositories/rudder-sdk-js...

Organization:      rudder-qa
Package manager:   npm
Target file:       examples/symfony/sample/package-lock.json
Project name:      package.json
Open source:       no
Project path:      /Volumes/Workspace/Repositories/rudder-sdk-js
Licenses:          enabled

✔ Tested 1 dependencies for known issues, no vulnerable paths found.

Next steps:
- Run `snyk monitor` to be notified about new related vulnerabilities.
- Run `snyk test` as part of your CI/test.


Tested 17 projects, 2 contained vulnerable paths.
```

- We can ignore the two unresolved dependencies as no path exists to fix them.

## Linear task (optional)

https://linear.app/rudderstack/issue/SDK-2451/fix-all-the-vulnerable-package-dependencies-js-sdk

## Cross Browser Tests

Please confirm you have tested for the following browsers:

- [ ] Chrome
- [ ] Firefox
- [ ] IE11

## Sanity Suite

- [ ] All sanity suite test cases pass locally

## Security

- [x] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

## Summary by CodeRabbit

- **New Features**
	- Enhanced GitHub Actions workflows with job dependencies for improved execution control.
- **Bug Fixes**
	- Updated cookie management logic in tests to reflect new domain and naming conventions.
- **Refactor**
	- Streamlined the `TestBook` class constructor for consistent test execution timing.
- **Chores**
	- Adjusted size limits in configuration for module formats to accommodate new requirements.
	- Updated ESLint configurations to include TypeScript-specific rules across multiple sample applications.
	- Updated Slack GitHub Action version in deployment workflows for improved performance.

### CODE REVIEW

File: packages/analytics-js/__fixtures__/msw.handlers.ts [hunk 1]
```diff
@@ -83,7 +83,7 @@ const handlers = [
       headers: {
         'Content-Type': 'application/json; charset=utf-8',
         'Set-Cookie':
-          'key1=value1; Path=/; Domain=example.com; Expires=Mon, 22 Apr 2025 10:55:28 GMT; SameSite=Lax',
+          'key1=value1; Path=/; Domain=dummy.dataplane.host.com; Expires=Mon, 22 Apr 2025 10:55:28 GMT; SameSite=Lax',
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-js/__fixtures__/msw.handlers.ts:86]: The change is only needed for test suites that actually made N/W requests.

File: packages/analytics-js/__tests__/components/userSessionManager/UserSessionManager.test.ts [hunk 2]
```diff
@@ -1689,7 +1689,7 @@ describe('User session manager', () => {
         state.storage.cookie.value = {
           maxage: 10 * 60 * 1000, // 10 min
           path: '/',
-          domain: 'example.com',
+          domain: 'dummy.dataplane.host.com',
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-js/__tests__/components/userSessionManager/UserSessionManager.test.ts:1692]: The change is only needed for test suites that actually made N/W requests.

File: examples/reactjs/ts/sample-app/src/reportWebVitals.ts [hunk 3]
```diff
@@ -1,13 +1,11 @@
...
-      getFID(onPerfEntry);
-      getFCP(onPerfEntry);
-      getLCP(onPerfEntry);
-      getTTFB(onPerfEntry);
+    import('web-vitals').then(({ onCLS, onFID, onFCP, onLCP, onTTFB }) => {
+      onCLS(onPerfEntry);
+      onFID(onPerfEntry);
+      onFCP(onPerfEntry);
+      onLCP(onPerfEntry);
+      onTTFB(onPerfEntry);
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply examples/reactjs/ts/sample-app/src/reportWebVitals.ts:8]: It is listed in the respective sample app's package.json file.

File: examples/reactjs/ts/sample-app/src/reportWebVitals.ts [hunk 4]
```diff
@@ -1,13 +1,11 @@
-import { ReportHandler } from 'web-vitals';
-
-const reportWebVitals = (onPerfEntry?: ReportHandler) => {
+const reportWebVitals = (onPerfEntry?: any) => {
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply examples/reactjs/ts/sample-app/src/reportWebVitals.ts:1]: It is not an issue for now.

File: examples/reactjs/hooks/sample-app/src/reportWebVitals.ts [hunk 5]
```diff
@@ -1,13 +1,11 @@
-import { ReportHandler } from 'web-vitals';
-
-const reportWebVitals = (onPerfEntry?: ReportHandler) => {
+const reportWebVitals = (onPerfEntry?: any) => {
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply examples/reactjs/hooks/sample-app/src/reportWebVitals.ts:1]: It is not an issue for now.

File: examples/reactjs/js/sample-app/src/reportWebVitals.js [hunk 6]
```diff
@@ -1,11 +1,11 @@
...
-      getFID(onPerfEntry);
-      getFCP(onPerfEntry);
-      getLCP(onPerfEntry);
-      getTTFB(onPerfEntry);
+    import('web-vitals').then(({ onCLS, onFID, onFCP, onLCP, onTTFB }) => {
+      onCLS(onPerfEntry);
+      onFID(onPerfEntry);
+      onFCP(onPerfEntry);
+      onLCP(onPerfEntry);
+      onTTFB(onPerfEntry);
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply examples/reactjs/js/sample-app/src/reportWebVitals.js:8]: It is listed in the respective sample app's package.json file.

**Review by MoumitaM — APPROVED** [review]

**Pull request merged**